tests/ui/variance/variance-regions-indirect.rs

```diff
@@ -6,23 +6,27 @@
 
 #[rustc_variance]
 enum Base<'a, 'b, 'c:'b, 'd> { //~ ERROR [-, +, o, *]
+    //~^ ERROR: `'d` is never used
     Test8A(extern "Rust" fn(&'a isize)),
     Test8B(&'b [isize]),
     Test8C(&'b mut &'c str),
 }
 
 #[rustc_variance]
 struct Derived1<'w, 'x:'y, 'y, 'z> { //~ ERROR [*, o, +, -]
+    //~^ ERROR: `'w` is never used
     f: Base<'z, 'y, 'x, 'w>
 }
```

Comment:
Is this correct? In all of these cases the lifetime that "isn't used" is mentioned in their `Base`. I'm guessing that the error triggers because the `'d` lifetime in `Base` isn't used, so transitively these lifetimes aren't used. It seems to me that we should either not emit the error in that case or point at `Base::'d` to explain what's happening.

Reply:
The error is correct and not new. It was just hidden by the variance debug "error" before